Gateway/Response/PaymentDetailsHandler.php

```diff
@@ -20,6 +51,27 @@ private function downloadPaymentFile($url)
         return curl_exec($ch);
     }
 
+    /**
+     * saves QR code in the media directory of magento.
+     * returns file path.
+     * @param \Magento\Payment\Gateway\Data\PaymentDataObject $paymentType
+     * @param string $qrCodeImage
+     * @return string
+     */
+    private function saveQr($paymentType, $qrCodeImage)
+    {
+        $filename = \Omise\Payment\Model\Config\Config::MEDIA_STORAGE_LOCATION;
+        $filename .= uniqid($paymentType."_");
+        if($this->_config->isSandboxEnabled()) {
```

Comment:
My feeling is that I'd rather use something like the file's mime type.  I don't think we should assume that the test mode file type will always be PNG.

Reply:
Actually I kinda wonder why we want to save the QR code in merchant's server?
How about just whenever user redirects to the `displayer.phtml` link, then, we just retrieve OmiseCharge object by a given order_id and display out the code that is in the Charge API body?